### PR TITLE
NMS-13256: Optimize kafka producer node generation

### PR DESCRIPTION
**Pull request by cgorantla**
Asynchronously update node topic without blocking sending events/alarms.

Also optimize hwtree retrieval.

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13256

### CODE REVIEW

**Comment by cgorantla** [comment]
This was verified with custom database dump which was taking 2 mins for a node update reduced to ~2 secs.

File: opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/HwEntityDaoHibernate.java [hunk 1]
```diff
@@ -55,6 +64,72 @@ public OnmsHwEntity findRootByNodeId(Integer nodeId) {
...
+
+    private void findChildren(HwEntity parent, List<HwEntity> hwEntityList) {
+
+        Set<HwEntity> children = hwEntityList.stream().filter(hwEntity -> {
+                    if (hwEntity.getParentId() != null) {
+                        return hwEntity.getParentId().equals(parent.getId());
+                    }
+                    return false;
+                }
+        ).collect(Collectors.toSet());
```
Comment (yangli2014) [inline opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/HwEntityDaoHibernate.java:106]: This section can be simplified as the following 
`Set<HwEntity> children = hwEntityList.stream().filter(hwEntity -> hwEntity.getParentId()!=null && hwEntity.getParentId().equals(parent.getId())).collect(Collectors.toSet());`

**Comment by yangli2014** [comment]
Nice job. Only minor comments

**Review by yangli2014 — APPROVED** [review]
After all of those failed tests being fixed it should be OK

**Comment by cgorantla** [comment]
> After all of those failed tests being fixed it should be OK

@yangli2014 Thanks for the review. Handled your comments. Tests also should be good now.

**Review by RangerRick — APPROVED** [review]
looks pretty straightforward

**Pull request merged**